js/modules/k6/grpc/client.go

```diff
@@ -0,0 +1,575 @@
+
+			// TODO(rogchap): Would be good to add support for custom RootCAs (self signed)
+
+			// (rogchap) we create a wrapper for transport credentials so that we can report
+			// on any TLS errors.
+			creds := transportCreds{
+				credentials.NewTLS(tlsCfg),
+				errc,
+			}
+			opts = append(opts, grpc.WithTransportCredentials(creds))
+		}
+
+		if isPlaintext {
+			opts = append(opts, grpc.WithInsecure())
+		}
+
+		dialer := func(ctx context.Context, addr string) (net.Conn, error) {
+			return state.Dialer.DialContext(ctx, "tcp", addr)
+		}
+		opts = append(opts, grpc.WithContextDialer(dialer))
+
+		ctx, cancel := context.WithTimeout(*ctxPtr, timeout)
+		defer cancel()
+
+		var err error
+		c.conn, err = grpc.DialContext(ctx, addr, opts...)
+		if err != nil {
+			errc <- err
+
+			return
+		}
+		close(errc)
+	}()
+
+	if err := <-errc; err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+// Invoke creates and calls a unary RPC by fully qualified method name
+//nolint: funlen,gocognit,gocyclo
+func (c *Client) Invoke(ctxPtr *context.Context,
+	method string, req goja.Value, params map[string]interface{}) (*Response, error) {
+	ctx := *ctxPtr
+	rt := common.GetRuntime(ctx)
+	state := lib.GetState(ctx)
+	if state == nil {
+		return nil, errInvokeRPCInInitContext
+	}
+
+	if c.conn == nil {
+		return nil, errors.New("no gRPC connection, you must call connect first")
+	}
+
+	if method == "" {
+		return nil, errors.New("method to invoke cannot be empty")
+	}
+
+	if method[0] != '/' {
+		method = "/" + method
+	}
+	md := c.mds[method]
+
+	if md == nil {
+		return nil, fmt.Errorf("method %q not found in file descriptors", method)
+	}
+
+	tags := state.CloneTags()
+	timeout := 60 * time.Second
+
+	ctx = metadata.NewOutgoingContext(ctx, metadata.New(nil))
+	for k, v := range params {
+		switch k {
+		case "headers":
+			rawHeaders, ok := v.(map[string]interface{})
+			if !ok {
+				return nil, errors.New("headers must be an object with key-value pairs")
+			}
+			for hk, kv := range rawHeaders {
+				// TODO(rogchap): Should we manage a string slice?
+				strVal, ok := kv.(string)
+				if !ok {
+					return nil, fmt.Errorf("header %q value must be a string", hk)
+				}
+				ctx = metadata.AppendToOutgoingContext(ctx, hk, strVal)
+			}
+		case "tags":
+			rawTags, ok := v.(map[string]interface{})
+			if !ok {
+				return nil, errors.New("tags must be an object with key-value pairs")
+			}
+			for tk, tv := range rawTags {
+				strVal, ok := tv.(string)
+				if !ok {
+					return nil, fmt.Errorf("tag %q value must be a string", tk)
+				}
+				tags[tk] = strVal
+			}
+		case "timeout":
+			var err error
+			switch t := v.(type) {
+			case string:
+				if timeout, err = time.ParseDuration(t); err != nil {
+					return nil, fmt.Errorf("unable to parse %q: %v", k, err)
+				}
+			case int64:
+				timeout = time.Duration(float64(t)) * time.Millisecond
+			case float64:
+				timeout = time.Duration(t) * time.Millisecond
```

Comment:
See https://play.golang.org/p/gsOWwDPPAV3 for the rationale

```suggestion
			case int64:
				timeout = time.Duration(t) * time.Millisecond
			case float64:
				timeout = time.Duration(t * float64(time.Millisecond))
```